### PR TITLE
Handle Background keyword correctly in cucumber

### PR DESCRIPTION
**Pull request by ono-max**
Currently, we are experiencing issue parsing the test reporting JSON file in the Cucumber framework. This PR addresses one of these issues, where we are not handling the Background keyword correctly. The Background keyword is similar to Before hooks in Cucumber. However, we are handling the steps in Background as an individual test case, which results in sending the same test path repeatedly. Therefore, we need to send a test case with combined steps in scenario and backgtround.

I'll address other issues in separate PRs.

### CODE REVIEW

File: launchable/test_runners/cucumber.py [hunk 1]
```diff
@@ -140,31 +141,27 @@ def parse_func(self, report_file: str) -> Generator[CaseEventType, None, None]:
...
-
-                if "failed" in statuses:
+                element_type = element.get("type", "")
+                if element_type == CucumberElementType.BACKGROUND.value:
+                    background_test_case_info = _extract_test_case_info_from_elements(element=element)
+                    continue
+
+                test_case_info = _extract_test_case_info_from_elements(element=element)
+                if background_test_case_info:
+                    test_case_info.statuses += background_test_case_info.statuses
```
Comment (ono-max) [inline launchable/test_runners/cucumber.py:157]: I'm not sure if we should concatenate steps in background_test_case_info. Please let me know if you have an opinion.

**Comment by Konboi** [comment]
@ono-max Please mention after fixing CI

**Comment by ono-max** [comment]
@Konboi 

The CI has passed.

File: launchable/test_runners/cucumber.py [hunk 2]
```diff
@@ -140,31 +141,27 @@ def parse_func(self, report_file: str) -> Generator[CaseEventType, None, None]:
             click.echo("Can't find test reports from {}. Make sure to confirm report file.".format(
                 report_file), err=True)
 
+        background_test_case_info = None
```
[the remark this answers is not in the document]
Comment (ono-max) [reply launchable/test_runners/cucumber.py:177]: [46903cc](https://github.com/launchableinc/cli/pull/640/commits/46903cc5f57e4ea1af19eab57a017f22878f8f4b)

File: launchable/test_runners/cucumber.py [hunk 3]
```diff
@@ -140,31 +141,27 @@ def parse_func(self, report_file: str) -> Generator[CaseEventType, None, None]:
...
-                        # ref:
-                        # https://github.com/cucumber/cucumber-ruby/blob/main/lib/cucumber/formatter/json.rb#L222
-                        duration = duration + result.get("duration", 0)
-                        statuses.append(result.get("status"))
-                        if result.get("error_message", None):
-                            stderr.append(result["error_message"])
-
-                if "failed" in statuses:
+                element_type = element.get("type", "")
+                if element_type == CucumberElementType.BACKGROUND.value:
```
[the remark this answers is not in the document]
Comment (ono-max) [reply launchable/test_runners/cucumber.py]: [3716d2b](https://github.com/launchableinc/cli/pull/640/commits/3716d2befbf6a69df52b33d42664a2c86733b744)

File: launchable/test_runners/cucumber.py [hunk 4]
```diff
@@ -175,11 +172,17 @@ def parse_func(self, report_file: str) -> Generator[CaseEventType, None, None]:
...
 
                 yield CaseEvent.create(
-                    test_path, duration / 1000 / 1000 / 1000, status, None, "\n".join(stderr), None, None)
+                    test_path,
+                    test_case_info.duration / 1000 / 1000 / 1000,
+                    status,
+                    None,
+                    "\n".join(test_case_info.stderr),
+                    None,
+                    None)
```
[the remark this answers is not in the document]
Comment (ono-max) [reply launchable/test_runners/cucumber.py]: [3903892](https://github.com/launchableinc/cli/pull/640/commits/39038926c6da81c0440465ca8acb36668d694142)

File: launchable/test_runners/cucumber.py [hunk 5]
```diff
@@ -216,3 +219,43 @@ def _create_file_candidate_list(file: str) -> List[str]:
...
+    def __init__(self, steps: Dict[str, str], duration: int, statuses: List[str], stderr: List[str]) -> None:
+        self.steps = steps
+        self.statuses = statuses
+        self.stderr = stderr
+        self.duration = duration
+
+
+def _extract_test_case_info_from_elements(element: Dict[str, List]) -> ElementTestCaseInfo:
+    steps = {}
+    duration = 0
```
[the remark this answers is not in the document]
Comment (ono-max) [reply launchable/test_runners/cucumber.py]: [bf914f9](https://github.com/launchableinc/cli/pull/640/commits/bf914f92752b85ecb7cf3f18f79f4ec04abb0dad)

**Review by Konboi — APPROVED** [review]

**Pull request merged**